### PR TITLE
Lemmoi/cp2k engine

### PR DESCRIPTION
**Pull request by lemmoi**
This PR implements most of the AbstractEngine methods for CP2K, allowing for construction from an input file with some basic validation, assignment of positions and velocities, and retrieving the system's atoms. There are also unit tests covering these.

Note that docstrings for overrided methods can be found in `abstract_engine.py` to avoid repeating documentation. Most code editors should be able to find these.

Out of scope:
 - Actual running of the engine (`run_shooting_point`). This will be its own PR
 - Some edge cases might cause problems, these are marked with TODOs

### CODE REVIEW

File: src/engines/CP2K_engine.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            self._atoms = [entry[0:2] for entry in self._get_coord()]
+
+        return self._atoms
+
+    def set_positions(self, positions: np.ndarray) -> None:
+        # Check positions are valid by passing to base class
+        super().set_positions(positions)
+
+        # coords stored as list of "El x y z" strings, same as CP2K .inp file
+        coords = self._get_coord()
```
Comment (ldgibson) [inline src/engines/CP2K_engine.py:34]: Where are these `coords` being stored outside of this method?

File: src/engines/CP2K_engine.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            return False, f"cp2k_inputs: {str(e)}"
+
+        # Otherwise let the base class validate
+        return super().validate_inputs(inputs)
+
+    async def run_shooting_point(self) -> ShootingResult:
+        pass
+
+    @property
+    def delta_t(self) -> float:
```
Comment (ldgibson) [inline src/engines/CP2K_engine.py:72]: Are these `delta_t` property methods intentionally empty and to be developed in later PRs?

Comment (lemmoi) [reply src/engines/CP2K_engine.py:72]: exactly

File: src/engines/CP2K_engine.py (hunk 1 above)
Comment (lemmoi) [reply src/engines/CP2K_engine.py:34]: Everything is being stored in the data structure `self.cp2k_inputs`, which is what the cp2k input parser creates. This is true for both coordinates and velocities. That function is just a helper to get to that specific section of the data structure.

Something i thought about was keeping these values in a separate array and only translating them into the `cp2k_inputs` when necessary, but we're almost always immediately write out a new inputs file after assigning new positions/velocities so there isn't much to gain by doing so.

**Pull request merged**